scripts/core/get-superdesk-api-implementation.tsx

```diff
@@ -118,6 +118,7 @@ export function getSuperdeskApiImplementation(
                 isPersonal: (article) => article.task == null || article.task.desk == null,
                 isLocked: (article) => article['lock_session'] != null,
                 isLockedByCurrentUser: (article) => lock.isLockedInCurrentSession(article),
+                hasMarkForUserPrivilege: () => privileges.userHasPrivileges({mark_for_user: 1}),
```

Comment:
API has `getOwnPrivileges` method - use that or add `hasPrivilege()` method. The core should not have any code related to a particular extension. All your changes for this task should be limited to files in the extension, except maybe if you want to add `hasPrivilege` method to the API.